core/templates/components/question-directives/question-editor/question-editor.component.ts

```diff
@@ -13,253 +13,258 @@
-        ctrl.directiveSubscriptions.unsubscribe();
-      };
+        });
+      }, () => {
+        // This callback is triggered when the Cancel button is
+        // clicked. No further action is needed.
+      });
+    }
+  }
+
+  saveInteractionId(displayedValue: string): void {
+    this._updateQuestion(() => {
+      this.stateEditorService.setInteractionId(cloneDeep(displayedValue));
+    });
+  }
+
+  saveInteractionAnswerGroups(newAnswerGroups: AnswerGroup[]): void {
+    this._updateQuestion(() => {
+      this.stateEditorService.setInteractionAnswerGroups(
+        cloneDeep(newAnswerGroups));
+    });
+  }
+
+  saveInteractionDefaultOutcome(newOutcome: Outcome): void {
+    this._updateQuestion(() => {
+      this.stateEditorService.setInteractionDefaultOutcome(
+        cloneDeep(newOutcome));
+    });
+  }
+
+  saveInteractionCustomizationArgs(
+      displayedValue: InteractionCustomizationArgs): void {
+    this._updateQuestion(() => {
+      this.stateEditorService.setInteractionCustomizationArgs(
+        cloneDeep(displayedValue));
+    });
+  }
+
+  saveNextContentIdIndex(displayedValue: number): void {
+    this._updateQuestion(() => {
+      let stateData = this.question?.getStateData();
+      stateData.nextContentIdIndex = cloneDeep(displayedValue);
+    });
+  }
+
+  saveSolution(displayedValue: Solution): void {
+    this._updateQuestion(() => {
+      this.stateEditorService.setInteractionSolution(
+        cloneDeep(displayedValue));
+    });
+
+    this.changeDetectionRef.detectChanges();
+  }
+
+  saveHints(displayedValue: Hint[]): void {
+    this._updateQuestion(() => {
+      this.stateEditorService.setInteractionHints(
+        cloneDeep(displayedValue));
+    });
+  }
+
+  saveInapplicableSkillMisconceptionIds(
+      displayedValue: string[]): void {
+    this.stateEditorService.setInapplicableSkillMisconceptionIds(
+      cloneDeep(displayedValue));
+    this.questionUpdateService.setQuestionInapplicableSkillMisconceptionIds(
+      this.question, displayedValue);
+  }
+
+  getStateContentPlaceholder(): string {
+    return 'Type your question here.';
+  }
+
+  getStateContentSaveButtonPlaceholder(): string {
+    return 'Save Question';
+  }
+
+  navigateToState(): void {
+    return;
+  }
+
+  addState(): void {
+    return;
+  }
+
+  recomputeGraph(): void {
+    return;
+  }
+
+  refreshWarnings(): void {
+    return;
+  }
+
+  _updateQuestion(updateFunction: Function): void {
+    this.questionChanged.emit();
+    this.questionUpdateService.setQuestionStateData(
+      this.question, updateFunction);
+  }
+
+  saveStateContent(displayedValue: SubtitledHtml): void {
+    // Show the interaction when the text content is saved, even if no
+    // content is entered.
+    this._updateQuestion(() => {
+      let stateData = this.question?.getStateData();
+      stateData.content = cloneDeep(displayedValue);
+      this.interactionIsShown = true;
+    });
+
+    // Shivam PTAL.
```

Comment:
Can you take the first pass yourself and then ask for a review?

Reply:
@ashutoshc8101 i have re-checked the pr.